### PR TITLE
[FR] Language and Locale as extensible enums

### LINKED ISSUE #16660: Expose locale parameter in method as an extensible enum instead of string
Needs to be per endpoint? looks like it

### PR DESCRIPTION
**Pull request by maririos**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/16660

This is from feedback .NET got in last arch board meeting. Because service team is currently exposing `Locale` and` Language` as generic to their service, I will follow the same in .NET.
There are plans in the future to have a consolidated method for all prebuilts and having specific enums per prebuilt model will not work.

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerLanguage.cs [hunk 1]
```diff
@@ -3,10 +3,10 @@
 
 using Azure.Core;
 
-namespace Azure.AI.FormRecognizer.Models
+namespace Azure.AI.FormRecognizer
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerLanguage.cs:6]: Decided to leave in main namespace as this is an input object to the methods in client. Follows the same as `FormContentType`

File: sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@
 - Renamed the model `Appearance` to `TextAppearance`.
 - Renamed the model `Style` to `TextStyle`.
 - Renamed the extensible enum `TextStyle` to `TextStyleName`.
+- Changed model type for `Locale` from `string` to `FormRecognizerLocale` for `RecognizeBusinessCardsOptions`, `RecognizeInvoicesOptions`, and `RecognizeReceiptsOptions`.
+- Changed model type for `Language` from `string` to `FormRecognizerLanguage` for `RecognizeContentOptions`.
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md:9]: They're definitely not wrong, but it was a bit hard to parse and understand these sentences, specially the first one. Maybe changing the prepositions to:
```
Changed model type of `Locale` from `string` to `FormRecognizerLocale` in `Recognize...`, `Recognize...`, and ...
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeBusinessCardsOptions.cs [hunk 3]
```diff
@@ -33,8 +33,8 @@ public RecognizeBusinessCardsOptions()
         public FormContentType? ContentType { get; set; }
 
         /// <summary>
-        /// Locale value. Supported locales include: en-AU, en-CA, en-GB, en-IN, en-US.
+        /// Set the locale information for the business card.
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeBusinessCardsOptions.cs:36]: Maybe `Sets the locale...` (like in `This property sets the locale...`). I always start these descriptions using the third person. Not sure if it's a standard or just a matter of preference, so it's up to you.

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeBusinessCardsOptions.cs:36]: In case you decide to change it, same comment in `RecognizeInvoiceOptions.Locale` and `RecognizeReceiptsOptions.Locale`.

**Review by kinelski — APPROVED** [review]
LGTM. Only left some discussions about English. 🇬🇧

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeInvoicesOptions.cs [hunk 4]
```diff
@@ -33,8 +33,8 @@ public RecognizeInvoicesOptions()
         public FormContentType? ContentType { get; set; }
 
         /// <summary>
-        /// Locale value. Supported locales include: en-US.
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeInvoicesOptions.cs:36]: would it make sense to keep this info? I know the user can look at options in the enum, but since not all those options are supported for invoice this sentence could be helpful.

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeInvoicesOptions.cs (hunk 4 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeInvoicesOptions.cs:36]: I was thinking about it, but every time the service makes an update to the supported locales I will have to come here an update the docstring...
I will bring it back for now and then we'll see

File: sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md]: Thank you!

**Comment by maririos** [comment]
@chidozieononiwu could you help me check why Linux keeps failing ?
